### PR TITLE
[PHP] Add ice 1.5.1

### LINKED ISSUE #2504: Add ice
Hi @mruz ,

Would you mind to add (or consent that we add) [ice](https://www.iceframework.org/
) in `here` ?

This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

### PR DESCRIPTION
**Pull request by mruz**
Closes #2504

### CODE REVIEW

File: php/ice/config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+framework:
+  website: iceframework.org
+  version: 1.5.1
```
Comment (waghanza) [inline php/ice/config.yaml:3]: Could you use `X.Y` format here ?

File: php/ice/config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+build_deps:
+  - bash
+
+before_build:
+  - wget -O /tmp/framework-1.5.1.tar.gz https://github.com/ice/framework/archive/1.5.1.tar.gz
```
Comment (waghanza) [inline php/ice/config.yaml:9]: could ice be installed with `pecl` or some other tool ?

File: php/ice/config.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  - cd /tmp/ && tar -xzf framework-1.5.1.tar.gz && cd framework-1.5.1 && bash install
+  - cp /tmp/ framework-1.5.1/tests/ci/ice.ini /usr/local/etc/php/conf.d/docker-php-ext-ice.ini
```
Comment (waghanza) [inline php/ice/config.yaml:11]: is this config file tuned for CI ? could it be tuned for a real-word app ?

File: php/ice/config.yaml (hunk 2 above)
Comment (mruz) [reply php/ice/config.yaml:9]: There is a packege, but only for the openSUSE linux https://software.opensuse.org/package/php7-ice, so it's needed to compile.

File: php/ice/config.yaml (hunk 3 above)
Comment (mruz) [reply php/ice/config.yaml:11]: It's just to turn the extension on:
```ini
extension=ice.so
```

Comment (waghanza) [reply php/ice/config.yaml:11]: Rather use an `echo extension=ice.so` then. 

I find it more explicit

File: php/ice/config.yaml (hunk 2 above)
Comment (waghanza) [reply php/ice/config.yaml:9]: In fact, I thought about a `pecl`, so as to have a tool doing compilation stuff fo us.
However, it is ok for this project. Do you have considered to distribute`ice` with `pecl` (it coul be easier when this project will run on cloud)

Comment (mruz) [reply php/ice/config.yaml:9]: Agree. I will try to add to the pecl repository.

File: php/ice/public/index.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+$di->loader
+    ->addNamespace('App', __DIR__ . '/../App')
+    ->register();
+
+$di->dispatcher
+    ->setNamespace('App');
+
+$di->router
+    ->setRoutes([
```
Comment (waghanza) [inline php/ice/public/index.php:13]: @mruz a route is missing : `['GET', '/']`

Comment (mruz) [reply php/ice/public/index.php:13]: https://github.com/the-benchmarker/web-frameworks/blob/684ac4a6088f640d11820dda79b5d1bfa1cc3c21/php/ice/public/index.php#L14
It's defined in the optional action `[/{action}]`, so not needed to define it again. Default action `index` will be used.

Comment (waghanza) [reply php/ice/public/index.php:13]: Strange then I've a server error on while requesting this endpoint

Comment (waghanza) [reply php/ice/public/index.php:13]: Could you rebase and push to trigger`CI` ?

Comment (waghanza) [reply php/ice/public/index.php:13]: Strange, travis do not start

Comment (mruz) [reply php/ice/public/index.php:13]: Screenshot_20200419_183556
I'm not sure why, but it says "Abuse detected" for all my commits https://github.com/travis-ci/travis-ci/issues/8626

Comment (waghanza) [reply php/ice/public/index.php:13]: I've seen that, I sent an email to travis' support. I'll probably create a `PR` in this repository, and include your commits :stuck_out_tongue:

Comment (waghanza) [reply php/ice/public/index.php:13]: this abuse was detected on my side, I donot know more, yet

**Pull request closed without merging**